src/query/src/parser.rs

```diff
@@ -54,25 +65,83 @@ impl QueryLanguageParser {
     }
 
     // TODO(ruihang): implement this method when parser is ready.
-    pub fn parse_promql(promql: &str) -> Result<QueryStatement> {
+    pub fn parse_promql(query: PromQuery) -> Result<QueryStatement> {
```

Comment:
Looks like `&PromQuery` is better for this function. It's not necessary to consume it.  Then we can change it in `execute_promql` too. Those stuffs 

```rust
let query_literal = format!("{query:?}");
```

can be removed.

Reply:
Have changed it to `&PromQuery`. But the `format!()` still exist for debug displaying the `PromQuery`